File: hw/chalan/u1/hw14Json/JavaScript/model/Farmer.js
export default class Farmer {
    #id;
    #name;
    #coopIds;

    constructor(id, name) {
        this.#id = id;
        this.#name = name;
        this.#coopIds = [];
    }

    getId() { return this.#id; }
    getName() { return this.#name; }
    getCoopIds() { return this.#coopIds; }

    addCoop(coopId) {
        if (!this.#coopIds.includes(coopId)) {
            this.#coopIds.push(coopId);
        }
    }

    toJSON() {
        return {
            id: this.#id,
            name: this.#name,
            coopIds: this.#coopIds
        };
    }

    static fromJSON(data) {
        const farmer = new Farmer(data.id, data.name);
        farmer.#coopIds = data.coopIds || [];
        return farmer;
    }

    toString() {
        return `Farmer{id: ${this.#id}, name: ${this.#name}, coops: ${this.#coopIds.length}}`;
    }
}

File: hw/chalan/u1/hw14Json/python/model/farmer.py
class Farmer:
    def __init__(self, farmer_id, name):
        self.farmer_id = farmer_id
        self.name = name
        self.coop_ids = []  # Lista de IDs de gallineros que posee
    
    def add_coop(self, coop_id):
        if coop_id not in self.coop_ids:
            self.coop_ids.append(coop_id)
    
    def to_dict(self):
        return {
            'farmer_id': self.farmer_id,
            'name': self.name,
            'coop_ids': self.coop_ids
        }
    
    @classmethod
    def from_dict(cls, data):
        farmer = cls(data['farmer_id'], data['name'])
        farmer.coop_ids = data['coop_ids']
        return farmer
    
    def __str__(self):
        return f"Farmer ID: {self.farmer_id}, Name: {self.name}, Coops: {len(self.coop_ids)}"